src/colmap/scene/synthetic.cc

```diff
@@ -277,6 +278,38 @@ void SynthesizeDataset(const SyntheticDatasetOptions& options,
       }
     }
 
+    if (options.use_prior_position) {
+      const Eigen::Vector3d noise(
+          RandomGaussian<double>(0, options.prior_position_stddev),
+          RandomGaussian<double>(0, options.prior_position_stddev),
+          RandomGaussian<double>(0, options.prior_position_stddev));
+
+      PosePrior noisy_prior(proj_center + noise,
+                            PosePrior::CoordinateSystem::CARTESIAN);
+
+      if (options.prior_position_stddev > 0.) {
+        noisy_prior.position_covariance = options.prior_position_stddev *
+                                          options.prior_position_stddev *
+                                          Eigen::Matrix3d::Identity();
+      } else {
+        noisy_prior.position_covariance = Eigen::Matrix3d::Identity();
+      }
+
+      if (options.use_geographic_coords_prior) {
+        static const GPSTransform gps_trans;
+
+        static const double lat0 = 47.37851943807808;
+        static const double lon0 = 8.549099927632087;
+        static const double alt0 = 451.5;
+
+        noisy_prior.position =
+            gps_trans.ENUToEll({noisy_prior.position}, lat0, lon0, alt0)[0];
+        noisy_prior.coordinate_system = PosePrior::CoordinateSystem::WGS84;
+      }
+
+      database->WritePosePrior(image_id, noisy_prior);
```

Comment:
I guess we could extend the synthetic_test.cc with a small check on consistent pairs of projection centers and pose priors (if the stddev is zero)?

Reply:
Done :+1: